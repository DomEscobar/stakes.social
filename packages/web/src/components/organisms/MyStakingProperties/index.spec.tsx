import React from 'react'
import { render } from '@testing-library/react'
import { MyStakingProperties } from '.'
import 'src/__mocks__/window/matchMedia.mock'
import { useGetAccountAddress, useBlockNumberStream, useGetBlock } from 'src/fixtures/wallet/hooks'

jest.mock('src/fixtures/dev-kit/hooks')
jest.mock('src/fixtures/wallet/hooks')
jest.mock('@dev/graphql')

describe(`${MyStakingProperties.name}`, () => {
  test('Snapshot', () => {
    ;(useGetAccountAddress as jest.Mock).mockResolvedValue('0x01234567890')
    ;(useBlockNumberStream as jest.Mock).mockResolvedValue('15987')
    ;(useGetBlock as jest.Mock).mockResolvedValue('15987')
    const component = render(<MyStakingProperties />)
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })
})
