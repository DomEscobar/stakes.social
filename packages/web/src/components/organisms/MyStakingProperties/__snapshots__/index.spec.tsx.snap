// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyStakingProperties Snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-1wzqvkc-0 fMjlez"
    >
      <h3
        class="ant-typography"
      >
        Staking Properties
      </h3>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Property
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Staking amount
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Staked day
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Elapsed days
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Withdrawable rewards
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Total earned
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <a
                            href="/0x01234567890"
                          >
                            authentication-id
                          </a>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            100
                             DEV
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span />
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span />
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            5000
                             DEV
                          </span>
                          <button
                            class="ant-btn ant-btn-primary"
                            type="button"
                          >
                            <span>
                              Withdraw
                            </span>
                          </button>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            10000
                             DEV
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="dy1yq4-0 dLtIEa"
                          >
                            <div
                              class="ant-space ant-space-horizontal ant-space-align-center"
                            >
                              <div
                                class="ant-space-item"
                                style="margin-right: 16px;"
                              >
                                <button
                                  class="ant-btn ant-btn-primary ant-btn-lg"
                                  type="button"
                                >
                                  <span>
                                    Cancel
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                                style="margin-right: 16px;"
                              >
                                <button
                                  class="ant-btn ant-btn-primary ant-btn-lg"
                                  disabled=""
                                  type="button"
                                >
                                  <span>
                                    Withdraw
                                  </span>
                                </button>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <div />
                              </div>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <a
                  class="ant-pagination-item-link"
                  disabled=""
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </a>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a>
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <a
                  class="ant-pagination-item-link"
                  disabled=""
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <h3
        class="ant-typography"
      >
        Summary
      </h3>
      <span>
        My Staking Properties: 
        1
      </span>
      <div
        class="ant-divider ant-divider-vertical"
        role="separator"
      />
      <span>
        Total Staking Amount: 
        100
         DEV
      </span>
      <div
        class="ant-divider ant-divider-vertical"
        role="separator"
      />
      <span>
        Total Earned: 
        0
         DEV
      </span>
    </div>
  </div>
</body>
`;
